db/model/model.js

```diff
@@ -17,12 +17,38 @@ exports.getArticleById = (id) => {
   }
   queryStr += " ORDER BY created_at";
 
-  return db.query(queryStr, queryValues).then(({ rows }) => {
-    if (rows.length === 0) {
-      return Promise.reject({ status: 404, msg: "not found" });
-    }
-    return rows;
-  });
+  return db
+    .query(queryStr, queryValues)
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+      return rows;
+    })
+    .then((rows) => {
+      return db
+        .query(`SELECT * FROM comments WHERE article_id = ${id}`)
+        .then(({ rows }) => {
+          console.log(rows);
+          return rows;
+        });
+    })
+    .then((comments) => {
+      return db
+        .query(`ALTER TABLE articles ADD comment_count INT DEFAULT 0`)
+        .then(() => {
+          const length = comments.length;
+          return db
+            .query(`UPDATE articles SET comment_count = $1`, [length])
+            .then(() => {
+              return db
+                .query(`SELECT * FROM articles WHERE article_id = $1`, [id])
+                .then(({ rows }) => {
+                  return rows;
+                });
+            });
+        });
```

Comment:
This is unfortunately quite heavily overthought. 

You should be able to amend the original query for articles, by adding another column to the query that counts up the comments using a COUNT. You will have to join the comments table to the articles table. 

We shouldn't be altering the data here - queries select stored data and present it to you in the format you request it - it doesn't alter what the tables look like. 

You also unfortunately have quite a few nested .thens here which is a separate issue, we should always chain .thens! 

I would definitely amend this to use a count and a join rather than the above 👍